src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -40,28 +42,24 @@ private void Dynamic_All(TraceEvent obj)
             // If we are paused, ignore the event. 
             // There's a potential race here between the two tasks but not a huge deal if we miss by one event.
             writer.ToggleStatus(pauseCmdSet);
-
-            if (obj.EventName.Equals("EventCounters"))
+            try
             {
-                IDictionary<string, object> payloadVal = (IDictionary<string, object>)(obj.PayloadValue(0));
-                IDictionary<string, object> payloadFields = (IDictionary<string, object>)(payloadVal["Payload"]);
+                if (obj.EventName.Equals("EventCounters"))
+                {
+                    IDictionary<string, object> payloadVal = (IDictionary<string, object>)(obj.PayloadValue(0));
+                    IDictionary<string, object> payloadFields = (IDictionary<string, object>)(payloadVal["Payload"]);
 
-                // If it's not a counter we asked for, ignore it.
-                if (!filter.Filter(obj.ProviderName, payloadFields["Name"].ToString())) return;
+                    // If it's not a counter we asked for, ignore it.
+                    if (!filter.Filter(obj.ProviderName, payloadFields["Name"].ToString())) return;
 
-                // There really isn't a great way to tell whether an EventCounter payload is an instance of 
-                // IncrementingCounterPayload or CounterPayload, so here we check the number of fields 
-                // to distinguish the two.
-                ICounterPayload payload;
-                if (payloadFields.ContainsKey("CounterType"))
-                {
-                    payload = payloadFields["CounterType"].Equals("Sum") ? (ICounterPayload)new IncrementingCounterPayload(payloadFields, _interval) : (ICounterPayload)new CounterPayload(payloadFields);
+                    ICounterPayload payload = payloadFields["CounterType"].Equals("Sum") ? (ICounterPayload)new IncrementingCounterPayload(payloadFields) : (ICounterPayload)new CounterPayload(payloadFields);
+                    writer.Update(obj.ProviderName, payload, pauseCmdSet);
                 }
-                else
-                {
-                    payload = payloadFields.Count == 6 ? (ICounterPayload)new IncrementingCounterPayload(payloadFields, _interval) : (ICounterPayload)new CounterPayload(payloadFields);
-                }
-                writer.Update(obj.ProviderName, payload, pauseCmdSet);
+            }
+            catch (Exception e)
```

Comment:
Writing catch(Exception) can make it quite challenging to diagnose bugs as it will prevent debuggers or error reporting tools like Watson from recognizing that an error occurred. There are a couple things we can do to improve this and I don't know all the details of what you are running into so I don't know which option(s) would be best:

a) In C# if you use:
````
try { bla }
catch(Exception e) when (DoSomething(e)) { }

bool DoSomething(Exception e)
{
    LogTheError(e);
    return false;
}
````
This would log the message but does not catch the exception (because the expression in the when clause returned false). Separately if you are going to do some generic error logging and then exit the app you'd prefer to do that logging in Main where it can be centralized rather than replicating similar error handling in different branches of the code. So most likely the code snippet would not go here.
b) If you narrow the catch clause to a certain Exception type you expect you can provide better error handling of those cases while not intervening in the exception types you didn't account for, aka bugs.
c) If this exception isn't working its way back to Main() one likely culprit is because the code never awaits for monitorTask that was started at line 219. Since you are polling for key-presses you might do something like:
````
while(true)
{
    Task completedTask = await Task.WhenAny(monitorTask, Task.Delay(250));
    if(completedTask == monitorTask)
    {
        await monitorTask; // this will finish instantly, and if monitorTask had an exception, this will throw it
        break;
    }
    ...
}